### PR TITLE
MDN CSS: Update regex in parse.pl

### PR DESCRIPTION
**Pull request by rasikapohankar**
Updates regex in parse.pl which handles titles having text inside brackets, to handle entries like `-moz-appearance (-webkit-appearance)` in addition to entries like `::before (:before)`.

###### People to notify 

@moollaza 

IA Page: https://duck.co/ia/view/mdn_css

Maintainer: @hchienjo

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by moollaza** [comment]
@rasikapohankar thanks a lot!

Can you please generate the new output.txt as well?

**Comment by rasikapohankar** [comment]
@moollaza Added the output file.

File: lib/fathead/mdn_css/parse.pl [hunk 1]
```diff
@@ -296,7 +296,7 @@ sub create_redirects {
 
     # Capture content inside parentheses
     # E.g. "::before (:before)"
-    if ( $title =~ m/(::.+) \((:.+)\)/ ) {
+    if ( $title =~ m/(.+) \((:.+|-webkit.+)\)/ ) {
```
Comment (moollaza) [inline lib/fathead/mdn_css/parse.pl]: @rasikapohankar I think we're going to need to handle the `-moz` and `-webkit` names with `()` in them slightly better. We're generating some odd redirects:

E.g. `-moz-tree-rowhover     R       :-moz-tree-row(hover)`

This should probably be: `-moz-tree-row hover     R       :-moz-tree-row(hover)`

Similarly, we're creating duplicates now as well:

- `-moz-placeholder       R       :-moz-placeholder`
- `-moz-placeholder       R       ::-moz-placeholder`

In those cases, we should actually generate a disambiguation.

Comment (moollaza) [inline lib/fathead/mdn_css/parse.pl]: @rasikapohankar my mistake, those redirects looks unrelated to these changes -- they must have made several changes to MDN recently 👍

**Comment by moollaza** [comment]
@rasikapohankar Thanks! This LGTM :+1:

File: lib/fathead/mdn_css/parse.pl (hunk 1 above)
Comment (rasikapohankar) [inline lib/fathead/mdn_css/parse.pl]: @moollaza I have a doubt, the RHS for `-moz-placeholder R` has `:-moz-placeholder` and `::-moz-placeholder`, which are different. Are they still considered duplicates?

Comment (moollaza) [inline lib/fathead/mdn_css/parse.pl]: @rasikapohankar yes, because we're redirecting the same query to two different queries, which doesn't work. Redirects must be 1-to-1 and unique. Any query that acts as a redirect cannot redirect to more than one article. Likewise if a query is specified as a redirect, it can't be used for anything else (like an article or a disambiguation).

Seeing as `:-moz-placeholder` and `::-moz-placeholder` are two different articles, and we can't determine which the user might want to see for the query `-moz-placeholder`, we should show a disambiguation result.

If there is only 1 of these cases then we may want to create a special case for this, otherwise, we should find a programmatic way of detecting these. We could even prepared a `disambigations.yaml` file that lists out all the disambiguations:

The LHS for every line in a Fathead output.txt must be unique. Any duplicates are technically overridden internally, but we shouldn't be generating any.

Comment (rasikapohankar) [inline lib/fathead/mdn_css/parse.pl]: @moollaza  got it :+1: Thanks a lot!
